plugins/inputs/ecs/types.go

```diff
@@ -54,22 +54,23 @@ func unmarshalTask(r io.Reader) (*Task, error) {
 }
 
 // docker parsers
-func unmarshalStats(r io.Reader) (map[string]types.StatsJSON, error) {
-	var statsMap map[string]types.StatsJSON
+func unmarshalStats(r io.Reader) (map[string]*types.StatsJSON, error) {
+	var statsMap map[string]*types.StatsJSON
 	err := json.NewDecoder(r).Decode(&statsMap)
 	return statsMap, err
```

Comment:
What about 
```suggestion
	return statsMap, json.NewDecoder(r).Decode(&statsMap)
```

Reply:
It works, but it looks a bit like a hack (the first argument returns a variable that is filled in the process of evaluating the second argument). 
For greater clarity, I changed it to a "more idiomatic" solution.